safesql.go

```diff
@@ -154,6 +180,70 @@ type QueryMethod struct {
 	Param    int
 }
 
+type Issue struct {
+	statement token.Position
+	ignored   bool
+}
+
+// CheckIssues checks lines to see if the line before or the current line has an ignore comment and marks those
+// statements that have the ignore comment on the current line or the line before
+func CheckIssues(lines []token.Position) ([]Issue, error) {
+	files := make(map[string][]token.Position)
+
+	for _, line := range lines {
+		files[line.Filename] = append(files[line.Filename], line)
+	}
+
+	issues := []Issue{}
+
+	for file, linesInFile := range files {
+		// ensure we have the lines in ascending order
+		sort.Slice(linesInFile, func(i, j int) bool { return linesInFile[i].Line < linesInFile[j].Line })
+
+		f, err := os.Open(file)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+		s := bufio.NewScanner(f)
+
+		currentLine := 0
+		for _, line := range linesInFile {
+			// check the line before the problematic statement first
+			potentialCommentLine := line.Line - 1
+
+			// if there are 2 statements back to back that are ignored then we don't want to check the previous so skip
+			// ie
+			// db.Query(query, args) //IsSqlSafe
```

Comment:
This should have some test cases to ensure that we handle the various cases, like:

- 2 consecutive lines ignored
- 1 line ignored, with a non-ignored line below
- etc.